lib/controller.dart

```diff
@@ -132,6 +138,32 @@ void loadDataFromFirebase() async {
   }
 }
 
+void loadGeoMaps() async {
```

Comment:
consider renaming this to `loadGeoMapsData`

If I understand the code correctly, it's only loading the data to make it available for rendering, but it's not doing the rendering itself. It would also make it a little bit more consistent with the other data loading method `loadDataFromFirebase` (though we could also consider renaming this one to `loadFirebaseData` for even more consistency, as the other way round: `loadDataFromGeoMaps` seems a bit inaccurate and awkward as a phrasing)

Reply:
Yes this is only loading the data. I like the consistency of renaming the firebase load to `loadFirebaseData`.  Fixed